python/pyspark/rdd.py

```diff
@@ -704,7 +704,16 @@ def pipe_objs(out):
                     out.write(s.encode('utf-8'))
                 out.close()
             Thread(target=pipe_objs, args=[pipe.stdin]).start()
-            return (x.rstrip(b'\n').decode('utf-8') for x in iter(pipe.stdout.readline, b''))
+
+            def check_return_code():
+                pipe.wait()
+                if pipe.returncode:
+                    raise Exception("Pipe function `%s' exited "
+                                    "with error code %d" % (command, pipe.returncode))
+                else:
+                    return None
```

Comment:
I think we should make `check_return_code` a generator, for example:
def check_return_code():
      # check return code
      for x in range(0):
         yield x